apps/etl/extraction/sources/base/handler.py

```diff
@@ -265,12 +265,12 @@ def _extraction_store_data(
         return extraction_object
 
     def _extraction_fetch_url(
-        self,
-        url: str,
-        headers: dict[str, typing.Any] | None = None,
+        self, url: str, headers: dict[str, typing.Any] | None = None, data: dict | None = None, method: str = "get"
```

Comment:
Let's use literal if we can (also use dict[str, typing.Any] instead of dict)
```suggestion
        self,
        url: str,
        headers: dict[str, typing.Any] | None = None,
        data: dict[str, typing.Any] | None = None,
        method: typing.Literal["get", "post"] = "get",
```